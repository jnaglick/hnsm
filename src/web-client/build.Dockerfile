FROM node:14
COPY tsconfig.base.json .
WORKDIR /src/web-client
CMD npm run build
