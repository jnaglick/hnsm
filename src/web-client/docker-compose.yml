version: "3.9"
services:
  web-client-build:
    build:
      context: .
      dockerfile: build.Dockerfile
    image: web-client-build
    volumes:
      - ./:/web-client
