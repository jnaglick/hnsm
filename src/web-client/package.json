{
  "name": "@hnsm/web-client",
  "version": "0.0.0",
  "description": "",
  "scripts": {
    "build": "webpack --mode production",
    "clean": "rm -rf node_modules/ && rm -rf dist/",
    "install:ci": "npm ci --no-optional",
    "start:dev": "webpack serve --live-reload --mode development",
    "test": "jest --config=../../jest.config.js --verbose --rootDir=."
  },
  "author": "",
  "license": "UNLICENSED",
  "dependencies": {
    "preact": "10.5.15"
  },
  "devDependencies": {
    "@types/jest": "27.0.3",
    "html-webpack-plugin": "5.4.0",
    "jest": "27.4.5",
    "ts-jest": "27.1.2",
    "ts-loader": "9.2.6",
    "tsconfig-paths-webpack-plugin": "3.5.2",
    "ts-node": "^10.7.0",
    "typescript": "4.4.4",
    "webpack": "5.59.1",
    "webpack-cli": "4.9.1",
    "webpack-dev-server": "4.4.0"
  }
}
