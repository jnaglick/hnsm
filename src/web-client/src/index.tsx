// eslint-disable-next-line @typescript-eslint/no-unused-vars
import { h, render, Component } from "preact";

class Hello extends Component {
  render() {
    return <div>Rendered, baby!!!!!</div>;
  }
}

render(<Hello />, document.body);
