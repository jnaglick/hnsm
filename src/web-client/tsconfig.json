{
  "extends": "../../tsconfig.base.json",

  "compilerOptions": {
    "target": "es2015" /* Specify ECMAScript target version: 'ES3' (default), 'ES5', 'ES2015', 'ES2016', 'ES2017', 'ES2018', 'ES2019' or 'ESNEXT'. */,
    "module": "es2015" /* Specify module code generation: 'none', 'commonjs', 'amd', 'system', 'umd', 'es2015', or 'ESNext'. */,
    
    "jsx": "react",
    "jsxFactory": "h" /* @jsx h */,
    "allowSyntheticDefaultImports": true,
    
    "baseUrl": "../.." /* to make TsconfigPathsPlugin not error ðŸ™„ */,
  },

  "exclude": ["node_modules", "dist", "**/*.spec.ts"]
}
