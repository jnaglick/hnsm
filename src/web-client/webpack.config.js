const HtmlWebpackPlugin = require("html-webpack-plugin");
const path = require("path");
const TsconfigPathsPlugin = require("tsconfig-paths-webpack-plugin");

const config = {
  entry: "./src/index.tsx",

  output: {
    path: path.resolve("./dist"),
    filename: "bundle.js",
  },

  resolve: {
    extensions: [".ts", ".tsx", ".js"],
    modules: [path.resolve("./node_modules"), path.resolve("./src")],
    plugins: [new TsconfigPathsPlugin()],
  },

  module: {
    rules: [
      {
        test: /\.(ts|tsx)$/,
        use: "ts-loader",
        exclude: /node_modules/,
      },
    ],
  },

  plugins: [
    new HtmlWebpackPlugin({
      title: "",
      filename: "index.html",
      template: "./src/index.html",
      chunks: [],
    }),
  ],

  target: "web",

  watchOptions: {
    ignored: /node_modules/,
  },

  devtool: "inline-source-map",

  devServer: {
    liveReload: true,
    port: 9000,
    client: {
      overlay: false,
    },
  },
};

module.exports = config;
