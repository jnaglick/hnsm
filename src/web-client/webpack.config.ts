import * as path from 'path';
import HtmlWebpackPlugin from 'html-webpack-plugin';
import TsconfigPathsPlugin from 'tsconfig-paths-webpack-plugin';

export const WEB_CLIENT_DIST_PATH = path.resolve(__dirname, "./dist");

const config = {
  entry: "./src/index.tsx",

  output: {
    path: WEB_CLIENT_DIST_PATH,
    filename: "bundle.js",
  },

  resolve: {
    extensions: [".ts", ".tsx", ".js"],
    modules: [path.resolve("./node_modules"), path.resolve("./src")],
    plugins: [new TsconfigPathsPlugin()],
  },

  module: {
    rules: [
      {
        test: /\.(ts|tsx)$/,
        use: "ts-loader",
        exclude: /node_modules/,
      },
    ],
  },

  plugins: [
    new HtmlWebpackPlugin({
      title: "",
      filename: "index.html",
      template: "./src/index.html",
      chunks: [],
    }),
  ],

  target: "web",

  watchOptions: {
    ignored: /node_modules/,
  },

  devtool: "inline-source-map",

  devServer: {
    liveReload: true,
    port: 9000,
    client: {
      overlay: false,
    },
  },
};

module.exports = config;
