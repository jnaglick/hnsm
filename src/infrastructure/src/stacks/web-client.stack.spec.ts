import * as cdk from "@aws-cdk/core";
import { SynthUtils } from "@aws-cdk/assert";

import { WebClientStack } from "./web-client.stack";

describe("WebClientStack", () => {
  it("should match snapshot", () => {
    // arrange
    const stack = new WebClientStack(
      new cdk.App({
        context: {
          stage: "test",
        },
      }),
      "WebClientStack"
    );

    // act
    const snapshot = SynthUtils.toCloudFormation(stack);

    // assert
    expect(snapshot).toMatchSnapshot();
  });
});
