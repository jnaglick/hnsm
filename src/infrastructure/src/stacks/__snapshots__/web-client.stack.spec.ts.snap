// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WebClientStack should match snapshot 1`] = `
Object {
  "Parameters": Object {
    "AssetParametersa3058ccb468d757ebb89df5363a1c20f5307c6911136f29d00e1a68c9b2aa7e8ArtifactHash238275D6": Object {
      "Description": "Artifact hash for asset \\"a3058ccb468d757ebb89df5363a1c20f5307c6911136f29d00e1a68c9b2aa7e8\\"",
      "Type": "String",
    },
    "AssetParametersa3058ccb468d757ebb89df5363a1c20f5307c6911136f29d00e1a68c9b2aa7e8S3BucketD1AD544E": Object {
      "Description": "S3 bucket for asset \\"a3058ccb468d757ebb89df5363a1c20f5307c6911136f29d00e1a68c9b2aa7e8\\"",
      "Type": "String",
    },
    "AssetParametersa3058ccb468d757ebb89df5363a1c20f5307c6911136f29d00e1a68c9b2aa7e8S3VersionKey93A19D70": Object {
      "Description": "S3 key for asset version \\"a3058ccb468d757ebb89df5363a1c20f5307c6911136f29d00e1a68c9b2aa7e8\\"",
      "Type": "String",
    },
    "AssetParameterse0659dd0cbb56ddc630c08444b84fb9f6d0054ce236e59b76340c02f447ecc80ArtifactHash8487259F": Object {
      "Description": "Artifact hash for asset \\"e0659dd0cbb56ddc630c08444b84fb9f6d0054ce236e59b76340c02f447ecc80\\"",
      "Type": "String",
    },
    "AssetParameterse0659dd0cbb56ddc630c08444b84fb9f6d0054ce236e59b76340c02f447ecc80S3Bucket3DEE379E": Object {
      "Description": "S3 bucket for asset \\"e0659dd0cbb56ddc630c08444b84fb9f6d0054ce236e59b76340c02f447ecc80\\"",
      "Type": "String",
    },
    "AssetParameterse0659dd0cbb56ddc630c08444b84fb9f6d0054ce236e59b76340c02f447ecc80S3VersionKey7870C312": Object {
      "Description": "S3 key for asset version \\"e0659dd0cbb56ddc630c08444b84fb9f6d0054ce236e59b76340c02f447ecc80\\"",
      "Type": "String",
    },
    "AssetParameterse9882ab123687399f934da0d45effe675ecc8ce13b40cb946f3e1d6141fe8d68ArtifactHashD9A515C3": Object {
      "Description": "Artifact hash for asset \\"e9882ab123687399f934da0d45effe675ecc8ce13b40cb946f3e1d6141fe8d68\\"",
      "Type": "String",
    },
    "AssetParameterse9882ab123687399f934da0d45effe675ecc8ce13b40cb946f3e1d6141fe8d68S3BucketAEADE8C7": Object {
      "Description": "S3 bucket for asset \\"e9882ab123687399f934da0d45effe675ecc8ce13b40cb946f3e1d6141fe8d68\\"",
      "Type": "String",
    },
    "AssetParameterse9882ab123687399f934da0d45effe675ecc8ce13b40cb946f3e1d6141fe8d68S3VersionKeyE415415F": Object {
      "Description": "S3 key for asset version \\"e9882ab123687399f934da0d45effe675ecc8ce13b40cb946f3e1d6141fe8d68\\"",
      "Type": "String",
    },
  },
  "Resources": Object {
    "BucketDeploymentAwsCliLayerB018DCE1": Object {
      "Properties": Object {
        "Content": Object {
          "S3Bucket": Object {
            "Ref": "AssetParameterse9882ab123687399f934da0d45effe675ecc8ce13b40cb946f3e1d6141fe8d68S3BucketAEADE8C7",
          },
          "S3Key": Object {
            "Fn::Join": Array [
              "",
              Array [
                Object {
                  "Fn::Select": Array [
                    0,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParameterse9882ab123687399f934da0d45effe675ecc8ce13b40cb946f3e1d6141fe8d68S3VersionKeyE415415F",
                        },
                      ],
                    },
                  ],
                },
                Object {
                  "Fn::Select": Array [
                    1,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParameterse9882ab123687399f934da0d45effe675ecc8ce13b40cb946f3e1d6141fe8d68S3VersionKeyE415415F",
                        },
                      ],
                    },
                  ],
                },
              ],
            ],
          },
        },
        "Description": "/opt/awscli/aws",
      },
      "Type": "AWS::Lambda::LayerVersion",
    },
    "BucketDeploymentCustomResourceCB7E2133": Object {
      "DeletionPolicy": "Delete",
      "Properties": Object {
        "DestinationBucketName": Object {
          "Ref": "WebClientBucketC2A4CF03",
        },
        "Prune": true,
        "ServiceToken": Object {
          "Fn::GetAtt": Array [
            "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756C81C01536",
            "Arn",
          ],
        },
        "SourceBucketNames": Array [
          Object {
            "Ref": "AssetParameterse0659dd0cbb56ddc630c08444b84fb9f6d0054ce236e59b76340c02f447ecc80S3Bucket3DEE379E",
          },
        ],
        "SourceObjectKeys": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                Object {
                  "Fn::Select": Array [
                    0,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParameterse0659dd0cbb56ddc630c08444b84fb9f6d0054ce236e59b76340c02f447ecc80S3VersionKey7870C312",
                        },
                      ],
                    },
                  ],
                },
                Object {
                  "Fn::Select": Array [
                    1,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParameterse0659dd0cbb56ddc630c08444b84fb9f6d0054ce236e59b76340c02f447ecc80S3VersionKey7870C312",
                        },
                      ],
                    },
                  ],
                },
              ],
            ],
          },
        ],
      },
      "Type": "Custom::CDKBucketDeployment",
      "UpdateReplacePolicy": "Delete",
    },
    "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756C81C01536": Object {
      "DependsOn": Array [
        "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756CServiceRoleDefaultPolicy88902FDF",
        "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756CServiceRole89A01265",
      ],
      "Properties": Object {
        "Code": Object {
          "S3Bucket": Object {
            "Ref": "AssetParametersa3058ccb468d757ebb89df5363a1c20f5307c6911136f29d00e1a68c9b2aa7e8S3BucketD1AD544E",
          },
          "S3Key": Object {
            "Fn::Join": Array [
              "",
              Array [
                Object {
                  "Fn::Select": Array [
                    0,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParametersa3058ccb468d757ebb89df5363a1c20f5307c6911136f29d00e1a68c9b2aa7e8S3VersionKey93A19D70",
                        },
                      ],
                    },
                  ],
                },
                Object {
                  "Fn::Select": Array [
                    1,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParametersa3058ccb468d757ebb89df5363a1c20f5307c6911136f29d00e1a68c9b2aa7e8S3VersionKey93A19D70",
                        },
                      ],
                    },
                  ],
                },
              ],
            ],
          },
        },
        "Handler": "index.handler",
        "Layers": Array [
          Object {
            "Ref": "BucketDeploymentAwsCliLayerB018DCE1",
          },
        ],
        "Role": Object {
          "Fn::GetAtt": Array [
            "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756CServiceRole89A01265",
            "Arn",
          ],
        },
        "Runtime": "python3.7",
        "Timeout": 900,
      },
      "Type": "AWS::Lambda::Function",
    },
    "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756CServiceRole89A01265": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756CServiceRoleDefaultPolicy88902FDF": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "s3:GetObject*",
                "s3:GetBucket*",
                "s3:List*",
              ],
              "Effect": "Allow",
              "Resource": Array [
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      "arn:",
                      Object {
                        "Ref": "AWS::Partition",
                      },
                      ":s3:::",
                      Object {
                        "Ref": "AssetParameterse0659dd0cbb56ddc630c08444b84fb9f6d0054ce236e59b76340c02f447ecc80S3Bucket3DEE379E",
                      },
                    ],
                  ],
                },
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      "arn:",
                      Object {
                        "Ref": "AWS::Partition",
                      },
                      ":s3:::",
                      Object {
                        "Ref": "AssetParameterse0659dd0cbb56ddc630c08444b84fb9f6d0054ce236e59b76340c02f447ecc80S3Bucket3DEE379E",
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
            Object {
              "Action": Array [
                "s3:GetObject*",
                "s3:GetBucket*",
                "s3:List*",
                "s3:DeleteObject*",
                "s3:PutObject*",
                "s3:Abort*",
              ],
              "Effect": "Allow",
              "Resource": Array [
                Object {
                  "Fn::GetAtt": Array [
                    "WebClientBucketC2A4CF03",
                    "Arn",
                  ],
                },
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      Object {
                        "Fn::GetAtt": Array [
                          "WebClientBucketC2A4CF03",
                          "Arn",
                        ],
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756CServiceRoleDefaultPolicy88902FDF",
        "Roles": Array [
          Object {
            "Ref": "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756CServiceRole89A01265",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "OriginAccessIdentityDF1E3CAC": Object {
      "Properties": Object {
        "CloudFrontOriginAccessIdentityConfig": Object {
          "Comment": "Allows CloudFront to reach the bucket",
        },
      },
      "Type": "AWS::CloudFront::CloudFrontOriginAccessIdentity",
    },
    "WebClientBucketC2A4CF03": Object {
      "DeletionPolicy": "Retain",
      "Properties": Object {
        "AccessControl": "Private",
      },
      "Type": "AWS::S3::Bucket",
      "UpdateReplacePolicy": "Retain",
    },
    "WebClientBucketPolicyB99ED3AC": Object {
      "Properties": Object {
        "Bucket": Object {
          "Ref": "WebClientBucketC2A4CF03",
        },
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "s3:GetObject*",
                "s3:GetBucket*",
                "s3:List*",
              ],
              "Effect": "Allow",
              "Principal": Object {
                "CanonicalUser": Object {
                  "Fn::GetAtt": Array [
                    "OriginAccessIdentityDF1E3CAC",
                    "S3CanonicalUserId",
                  ],
                },
              },
              "Resource": Array [
                Object {
                  "Fn::GetAtt": Array [
                    "WebClientBucketC2A4CF03",
                    "Arn",
                  ],
                },
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      Object {
                        "Fn::GetAtt": Array [
                          "WebClientBucketC2A4CF03",
                          "Arn",
                        ],
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::S3::BucketPolicy",
    },
    "WebClientDistribution282DC30F": Object {
      "Properties": Object {
        "DistributionConfig": Object {
          "DefaultCacheBehavior": Object {
            "CachePolicyId": "658327ea-f89d-4fab-a63d-7e88639e58f6",
            "Compress": true,
            "TargetOriginId": "WebClientStackWebClientDistributionOrigin1064A2D61",
            "ViewerProtocolPolicy": "allow-all",
          },
          "DefaultRootObject": "index.html",
          "Enabled": true,
          "HttpVersion": "http2",
          "IPV6Enabled": true,
          "Origins": Array [
            Object {
              "DomainName": Object {
                "Fn::GetAtt": Array [
                  "WebClientBucketC2A4CF03",
                  "RegionalDomainName",
                ],
              },
              "Id": "WebClientStackWebClientDistributionOrigin1064A2D61",
              "S3OriginConfig": Object {
                "OriginAccessIdentity": Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      "origin-access-identity/cloudfront/",
                      Object {
                        "Ref": "OriginAccessIdentityDF1E3CAC",
                      },
                    ],
                  ],
                },
              },
            },
          ],
        },
      },
      "Type": "AWS::CloudFront::Distribution",
    },
  },
}
`;
